samples/go/csv/read.go

```diff
@@ -146,17 +131,18 @@ func ReadToMap(r *csv.Reader, headersRaw []string, pkIdx int, kinds KindSlice, v
 
 		fieldIndex := 0
 		var pk types.Value
-		fields := make(types.ValueSlice, len(headers))
+		fields := make(types.ValueSlice, len(headersRaw))
 		for x, v := range row {
-			if x == pkIdx {
-				pk, err = StringToValue(v, pkKind)
-			} else if fieldIndex < len(headers) {
+			if fieldIndex < len(headersRaw) {
```

Comment:
should always be less, except if the csv file has a different number of fields that the header row indicated.